model/jpa/src/main/java/org/keycloak/models/jpa/session/PersistentUserSessionEntity.java

```diff
@@ -35,6 +35,9 @@
         @NamedQuery(name="deleteUserSessionsByRealm", query="delete from PersistentUserSessionEntity sess where sess.realmId = :realmId"),
         @NamedQuery(name="deleteUserSessionsByUser", query="delete from PersistentUserSessionEntity sess where sess.userId = :userId"),
         @NamedQuery(name="deleteExpiredUserSessions", query="delete from PersistentUserSessionEntity sess where sess.realmId = :realmId AND sess.offline = :offline AND sess.lastSessionRefresh < :lastSessionRefresh"),
+        @NamedQuery(name="deleteExpiredUserSessionsCrossRealm", query="delete from PersistentUserSessionEntity sess where"
+                + " sess.realmId in (select realm.id from RealmEntity realm where realm.offlineSessionIdleTimeout = :offlineSessionIdle)"
```

Comment:
"nitpicking" just for consistency 
```suggestion
                + " sess.realmId IN (select realm.id from RealmEntity realm where realm.offlineSessionIdleTimeout = :offlineSessionIdle)"
```

Reply:
Thanks, updated